### PR TITLE
Test for effect

### PR DESCRIPTION
**Pull request by mzgoddard**
### Proposed changes

- Add node-tap dependency for test running
- Shape test Effect subclasses
- Add quality of life assertions to Effect.connect

### Reason for changes

- First tests for scratch-audio
- Effect's connecting can be a little hard to follow, test it to make sure its correct
- Effect.connect assertions make sure you don't make easy errors connecting Effects together

### CODE REVIEW

File: test/__mocks__/AudioNode.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class AudioNodeMock {
```
Comment (gnarf) [inline test/__mocks__/AudioNode.js:1]: I think you should go through the other mock classes and make sure their class name ends with `Mock` this will be useful in debugging errors some day I'm sure.

File: test/__mocks__/AudioNode.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    disconnect () {
+        if (this.connected !== null) {
+            this.connectedFrom = this.connectedFrom.filter(connection => connection !== this);
+        }
+        this.connected = null;
+    }
+
+    _test (test, message, depth = 0) {
```
Comment (gnarf) [inline test/__mocks__/AudioNode.js:22]: Love the method body, but meh on the name, perhaps: `testConnectionPropagation` or oder or something?

**Review by gnarf — APPROVED** [review]
Overall :+1: this is awesome, I think that `_test` feels a bit awkward to read in the test code, perhaps rename?

**Review by rschamp — APPROVED** [review]
Looks great! And thanks for parallelizing Travis too!
